crates/core/tedge_mapper/src/main.rs

```diff
@@ -101,9 +101,11 @@ async fn main() -> anyhow::Result<()> {
     };
     set_log_level(log_level);
 
-    // Run only one instance of a mapper
-    let run_dir: PathBuf = config.query(RunPathSetting)?.into();
-    let _flock = check_another_instance_is_not_running(&mapper_opt.name.to_string(), &run_dir)?;
+    // Run only one instance of a mapper (if enabled)
+    if config.query(LockFilesSetting)?.is_set() {
+        let run_dir: PathBuf = config.query(RunPathSetting)?.into();
+        let _flock = check_another_instance_is_not_running(&mapper_opt.name.to_string(), &run_dir)?;
```

Comment:
Since the lock is released on drop, you have to use here the same trick as for the agent, i.e. to hold a reference to the `Some(Lock)`.

Reply:
Fixed